merino-adm/src/remote_settings/reqwest_client.rs

```diff
@@ -11,23 +11,29 @@ use remote_settings_client::client::net::{
     Url as RsUrl,
 };
 use reqwest::{header::CONTENT_TYPE, Method, Response};
+use std::time::Duration;
 
 /// An remote-settings-client HTTP client that uses Reqwest.
 #[derive(Debug)]
 pub struct ReqwestClient {
     /// The client that will be used to make http requests.
     reqwest_client: reqwest::Client,
+    /// The HTTP timeout in seconds.
+    http_timeout: Duration,
 }
 
 impl ReqwestClient {
     /// Instantiate a new Reqwest client to perform HTTP requests.
-    pub fn try_new() -> Result<ReqwestClient, Error> {
+    pub fn try_new(http_timeout: Duration) -> Result<ReqwestClient, Error> {
         let reqwest_client = reqwest::Client::builder()
             // Disable the connection pool to avoid the IncompleteMessage errors.
```

Comment:
You can instead set it on the builder here instead of on every request

```suggestion
            .timeout(http_timeout)
            // Disable the connection pool to avoid the IncompleteMessage errors.
```